src/test/ui/traits/static-method-generic-inference.stderr

```diff
@@ -9,8 +9,8 @@ LL |     let _f: base::Foo = base::HasNew::new();
    |
 help: use the fully-qualified path to the only available implementation
    |
-LL |     let _f: base::Foo = base::<Foo as HasNew>::new();
-   |                               +++++++       +
+LL |     let _f: base::Foo = <Foo as base::HasNew>::new();
```

Comment:
This is an improvement, but for it to actually compile it needs a path to Foo and type params for HasNew:

`let _f: base::Foo = <base::Foo as base::HasNew<_>>::new();`

I can spin this off into a new issue (or two new issues) if you prefer.

Reply:
You can make a new issue for this. Thanks for pointing it out.

Reply:
Reported the missing path as #117623. The missing type params is probably okay, since the problem is in the original code rather than being introduced by the hint.